igc.cc

```diff
@@ -348,6 +361,60 @@ void IgcFormat::read()
       }
       prev_tod = tod;
 
+      /*
+       * Parse any extension data present. If no extensions are used (unlikely,
+       * but possible in the case of homebrew flight recorders), then skip the
+       * whole thing.
+      */
+
+      // TODO: A switch would be better than an if ladder...
+      if (igc_metadata.flags.has_igc_exts){
+        auto* fsdata = new igc_fsdata;
+        QHash<QString, short>::const_iterator i;
+        for (i = ext_types_hash.constBegin(); i != ext_types_hash.constEnd(); ++i) {
+          QStringView ext_record_type = QStringView(i.key().mid(4,3)).toString();
+          short start = igc_metadata.extension(ext_record_type.toString())->start;
+          short end = igc_metadata.extension(ext_record_type.toString())->end;
+          int len = end - start + 1;  // The difference results in a fencepost error
+          extension = igc_metadata.extension(ext_record_type.toString());
+          ext_data = QStringView(ibuf_q).mid(igc_metadata.extension(ext_record_type.toString())->start, len);
+          if (ext_record_type.toString() == "ENL") {
+            fsdata->enl = ext_data.toInt();
+            fsdata->igc_fs_flags.enl = fsdata->igc_fs_flags.enl || true;
+          } else if (ext_record_type.toString() == "TAS") {
+            fsdata->tas = ext_data.toInt();
+            fsdata->igc_fs_flags.tas = fsdata->igc_fs_flags.tas || true;
+          } else if (ext_record_type.toString() == "VAT") {
+            fsdata->vat = ext_data.toInt();
+            fsdata->igc_fs_flags.vat = fsdata->igc_fs_flags.vat || true;
+          } else if (ext_record_type.toString() == "OAT") {
```

Comment:
I think OAT, Outside air temperature (Celsius), after scaling, can go into Waypoint::temperature, a float.  (pres_wpt->set_temperature(ext_data.toInt()/scale).  It looks like there is an implied decimal point if the OAT field has more than three characters, see spec A2.4.1.  Therefore, we don't need igc_fs_flags.oat.